slick/src/main/scala/slick/lifted/ExtensionMethods.scala

```diff
@@ -43,6 +43,8 @@ trait ColumnExtensionMethods[B1, P1] extends Any with ExtensionMethods[B1, P1] {
 
   def in[P2, R, C[_]](e: Query[Rep[P2], _, C])(implicit om: o#arg[B1, P2]#to[Boolean, R]) =
     om.column(Library.In, n, e.toNode)
+  def in[P2, R](first: Rep[P2], second: Rep[P2], rest: Rep[P2]*)(implicit om: o#arg[B1, P2]#to[Boolean, R]) =
```

Comment:
What is first and second? Why does there have to be at least two?

Reply:
Ah, sorry, I thought I'd already explained [in the comment above](https://github.com/slick/slick/pull/1998#issuecomment-469267785) but maybe I wasn't clear:

I've chosen to stick with the name `in` for this function, since it maps directly to the keyword in SQL (I'm not wedded to this, although it would be nice to stay close to the SQL if possible). My first attempt was to write a function with the signature:

```scala
def in[P2, R](es: Rep[P2])...
```

but this doesn't work: the already-existing `in` method has a signature of `def in(e: Query)` and, since `Query <: Rep` the compiler can't tell the difference between the two after erasure.

So, by using `first` and `second` I force the end-user to put *at least* two lifted values inside the `in` clause, so that the compiler can tell the function signatures apart, and select the right one; having `rest` as varargs then allows arbitrarily many additional lifted values to be used.

It's possible they only want to use one value, but writing `where foo in (bar)` in SQL doesn't make a lot of sense since it's identical to `where foo = bar`, so we can use slick's `===` instead.